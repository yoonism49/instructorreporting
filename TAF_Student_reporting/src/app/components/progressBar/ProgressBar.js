// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';

const ProgressBar = ({
  barSize,
  barStriped,
  barStyle,
  active,
  valueNow,
  valueMin,
  valueMax,
  screenReadersText
}) => {
  return (
    <div
      className={
        cx({
          progress:       true,
          'progress-xxs': barSize === 'xxs',
          'progress-xs':  barSize === 'xs',
          'progress-sm':  barSize === 'sm',
          'active':       active
        })
      }>
      <div
        className={
          cx({
            'progress-bar':         true,
            'progress-bar-striped': barStriped,
            'progress-bar-warning': barStyle === 'warning',
            'progress-bar-primary': barStyle === 'primary',
            'progress-bar-success': barStyle === 'success',
            'progress-bar-danger':  barStyle === 'danger'
          })
        }
        role="progressbar"
        aria-valuenow={valueNow}
        aria-valuemin={valueMin}
        aria-valuemax={valueMax}
        style={{width: `${valueNow}%`}}>
        <span className="sr-only">
          {screenReadersText}
        </span>
      </div>
    </div>
  );
};

ProgressBar.propTypes = {
  barSize:    PropTypes.oneOf(['xxs', 'xs', 'sm', 'progress']),
  barStriped: PropTypes.bool,
  barStyle:   PropTypes.oneOf(['primary', 'success', 'warning', 'danger']),
  active:     PropTypes.bool,
  valueNow:   PropTypes.number,
  valueMin:   PropTypes.number,
  valueMax:   PropTypes.number,
  screenReadersText: PropTypes.string
};

ProgressBar.defaultProps = {
  barSize:    'progress',
  barStriped: false,
  barStyle:   'primary',
  active:     false,
  valueNow:   0,
  valueMin:   0,
  valueMax:   100,
  screenReadersText: 'progress'
};

export default ProgressBar;
