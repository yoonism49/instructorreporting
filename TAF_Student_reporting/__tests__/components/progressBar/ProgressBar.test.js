
'use strict';

import React              from 'react';
import ProgressBar        from '../../../src/app/components/progressBar/ProgressBar';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('ProgressBar component', () => {
  it('renders as expected', () => {
    const mockProps = {
      barSize: 'xs',
      barStriped: true,
      barStyle: 'primary',
      active: false,
      valueNow: 10,
      valueMin: 0,
      valueMax: 50,
      screenReadersText: 'progress'
    };
    const component = renderer.create(
      <div>
        <ProgressBar {...mockProps } />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
