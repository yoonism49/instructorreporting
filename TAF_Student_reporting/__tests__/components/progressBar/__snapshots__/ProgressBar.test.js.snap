// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders as expected 1`] = `
<div>
  <div
    className="progress progress-xs"
  >
    <div
      aria-valuemax={50}
      aria-valuemin={0}
      aria-valuenow={10}
      className="progress-bar progress-bar-striped progress-bar-primary"
      role="progressbar"
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <span
        className="sr-only"
      >
        progress
      </span>
    </div>
  </div>
</div>
`;
